src/Api/Test/ClientTest.php

```diff
@@ -0,0 +1,325 @@
+        $this->assertHydraError('Script "'.self::UNEXISTING_ID.'" does not exist');
+    }
+
+    public function testCreateClientUnexistentPreScript(): void
+    {
+        $httpClient = $this->authenticate();
+        $clientName = $this->createClientName();
+        $clientJson = ClientMother::withPreScripts($clientName, [self::UNEXISTING_ID]);
+        $this->postClient($httpClient, $clientJson);
+        $this->assertResponseStatusCodeSame(400);
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertHydraError('Script "'.self::UNEXISTING_ID.'" does not exist');
+    }
+
+    public function testDeleteClient(): void
+    {
+        $httpClient = $this->authenticate();
+        $iri = $this->findIriBy(Client::class, [
+            'name' => 'client_to_delete'
+        ]);
+        $response = $httpClient->request('DELETE', $iri);
+        $this->assertResponseIsSuccessful();
+        $response = $httpClient->request('GET', $iri);
+        $this->assertResponseStatusCodeSame(404);
+    }
+
+    public function testDeleteClientNotFound(): void
+    {
+        $httpClient = $this->authenticate();
+        $response = $httpClient->request('DELETE', 'api/clients/'.self::UNEXISTING_ID);
+        $this->assertResponseStatusCodeSame(404);
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertHydraError('The client "'.self::UNEXISTING_ID.'" does not exist.');
+    }
+
+    public function testGetClients(): void
+    {
+        $httpClient = $this->authenticate();
+        $response = $httpClient->request('GET', '/api/clients');
+        $this->assertResponseIsSuccessful();
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertJsonContains([
+        '@context' => '/api/contexts/Client',
+        '@type' => 'hydra:Collection',
+        '@id' => '/api/clients',
+        ]);
+    }
+
+    public function testGetClientsUnauthenticated(): void
+    {
+        $response = static::createClient()->request('GET', '/api/clients');
+        $this->assertResponseStatusCodeSame(401);
+    }
+
+    public function testGetNonexistentClient(): void
+    {
+        $httpClient = $this->authenticate();
+        $response = $httpClient->request('GET', '/api/clients/'.self::UNEXISTING_ID);
+        $this->assertResponseStatusCodeSame(404);
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertHydraError('The client "'.self::UNEXISTING_ID.'" does not exist.');
+    }
+
+    public function testGetClient(): void
+    {
+        $httpClient = $this->authenticate();
+        $clientName = $this->createClientName();
+        $clientJson = ClientMother::named($clientName);
+        $this->postClient($httpClient, $clientJson);
+        $iri = $this->findIriBy(Client::class, [
+            'name' => $clientName
+        ]);
```

Comment:
Not bad, but can we have an existing fixture loaded client and just
`ClientMother::getExistingClientId();`